pkg/controller/certificaterequests/util/reporter.go

```diff
@@ -59,6 +59,26 @@ func (r *Reporter) Failed(cr *cmapi.CertificateRequest, err error, reason, messa
 
 }
 
+func (r *Reporter) Denied(cr *cmapi.CertificateRequest) {
+	// Set the FailureTime to c.clock.Now(), only if it has not been already set.
+	if cr.Status.FailureTime == nil {
+		nowTime := metav1.NewTime(r.clock.Now())
+		cr.Status.FailureTime = &nowTime
+	}
+
+	message := "The CertificateRequest was denied by an approval controller"
+
+	// If RequestDenied condition not already set then fire a RequestDenied
+	// Event. This is to reduce strain on the API server and avoid rate limiting
+	// ourselves for Event creation.
+	if apiutil.CertificateRequestReadyReason(cr) != cmapi.CertificateRequestReasonDenied {
+		r.recorder.Event(cr, corev1.EventTypeWarning, cmapi.CertificateRequestReasonDenied, message)
```

Comment:
As noted above, I think maybe we are better to _not_ have this event? Not too sure.. seems like we will end out with duplicate events

Reply:
(because the approval controller is likely to make a similar event)